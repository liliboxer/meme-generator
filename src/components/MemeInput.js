import React from 'react';
import PropTypes from 'prop-types';

function MemeInput({ top, bottom, handleChange }) {
  return (
    <>
      <form>
        <input name="top" value={top} placeholder="top text" onChange={(event) => handleChange(event)}></input>
        <input name="bottom" value={bottom}placeholder="bottom text" onChange={(event) => handleChange(event)}></input>
      </form>
    </>
  );
}

MemeInput.propTypes = {
  top: PropTypes.string.isRequired,
  imageURL: PropTypes.string.isRequired,
  bottom: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export default MemeInput;
