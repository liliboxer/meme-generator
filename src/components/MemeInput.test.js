import React from 'react';
import { shallow } from 'enzyme';
import MemeInput from './MemeInput';

describe('MemeInput component', () => {
  it('renders MemeInput', () => {
    const wrapper = shallow(<MemeInput />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
