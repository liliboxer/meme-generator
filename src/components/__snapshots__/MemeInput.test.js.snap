// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemeInput component renders MemeInput 1`] = `
<Fragment>
  <form>
    <input
      name="top"
      onChange={[Function]}
      placeholder="top text"
      type="text"
    />
    <input
      name="bottom"
      onChange={[Function]}
      placeholder="bottom text"
      type="text"
    />
  </form>
</Fragment>
`;
